packages/boot/test/bootstrapTests/ec-contract-upgrade.test.ts

```diff
@@ -0,0 +1,601 @@
+import { test as anyTest } from '@agoric/zoe/tools/prepare-test-env-ava.js';
+import { TestFn } from 'ava';
+import { eventLoopIteration } from '@agoric/internal/src/testing-utils.js';
+import {
+  makeAgoricNamesRemotesFromFakeStorage,
+  slotToBoardRemote,
+  unmarshalFromVstorage,
+} from '@agoric/vats/tools/board-utils.js';
+import { makeMarshal } from '@endo/marshal';
+
+import { makeSwingsetTestKit } from '../../tools/supports.js';
+import { makeWalletFactoryDriver } from '../../tools/drivers.js';
+
+const wallets = {
+  gov1: 'agoric1gx9uu7y6c90rqruhesae2t7c2vlw4uyyxlqxrx',
+  gov2: 'agoric1d4228cvelf8tj65f4h7n2td90sscavln2283h5',
+  gov3: 'agoric14543m33dr28x7qhwc558hzlj9szwhzwzpcmw6a',
+  gov4: 'agoric13p9adwk0na5npfq64g22l6xucvqdmu3xqe70wq',
+  gov5: 'agoric1el6zqs8ggctj5vwyukyk4fh50wcpdpwgugd5l5',
+  gov6: 'agoric1zayxg4e9vd0es9c9jlpt36qtth255txjp6a8yc',
+};
+
+const latestQuestionKey = `published.committees.Economic_Committee.latestQuestion`;
+const lastOutcomeKey = `published.committees.Economic_Committee.latestOutcome`;
+const highPrioritySenderKey = 'highPrioritySenders';
+
+const PROPOSAL_INV = 'old_proposal_invitation';
+const VOTER_INV = 'old_voter_invitation';
+const NEW_VOTER_INV = 'new_voter_invitation';
```

Comment:
I see from below that these are offer ids. That was hard to tell from these names.

Also, since "old" and "new" are ambiguous, let's stick to "outgoing", "continuing" and "incoming" membership.

```suggestion
const PROPOSAL_INV_OFFER_ID = 'outgoing_proposal_invitation';
const VOTER_INV_OFFER_ID = 'outgoing_voter_invitation';
const INCOMING_VOTER_INV_OFFER_ID = 'incoming_voter_invitation';
```

or perhaps:

```suggestion
const offerIds = {
  propose: { outgoing: 'outgoing_propose_invitation' },
  vote: { outgoing: 'outgoing_vote_invitation', incoming: 'incoming_vote_invitation' },
};
```

You might drop `_invitation`; that seem like it should be represented by `source: 'purse'` in the offer spec.